.github/workflows/go.yml

```diff
@@ -203,12 +203,23 @@ jobs:
       uses: actions/checkout@v3
     - name: Run verify scripts
       run: make verify
-    - name: Checking for broken Markdown links
+    - name: Checking for broken Markdown links for main branch
+      if: ${{ github.base_ref == 'main' }}
       uses: gaurav-nelson/github-action-markdown-link-check@v1
       with:
         folder-path: './docs'
         file-path: './README.md, ./CHANGELOG.md, ./CONTRIBUTING.md, ./GOVERNANCE.md, ./MAINTAINERS.md, ./ROADMAP.md, ./SECURITY.md'
         config-file: 'hack/.md_links_config.json'
+    - name: Checking for broken Markdown links for release branch
+      if: ${{ startsWith(github.base_ref, 'release-') }}
+      uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        # Check modified files only as there may be stale links in previous CHANGELOGs
+        check-modified-files-only: yes
+        base-branch: ${{ github.base_ref }}
+        folder-path: './docs, ./CHANGELOG'
+        file-path: './README.md, ./CHANGELOG.md, ./CONTRIBUTING.md, ./GOVERNANCE.md, ./MAINTAINERS.md, ./ROADMAP.md, ./SECURITY.md'
+        config-file: 'hack/.md_links_config.json'
```

Comment:
Maybe the previous step could always run, and this step would only run for `startsWith(github.base_ref, 'release-')` and only check the `./CHANGELOG` directory? This way you don't need to repeat the whole list of files.

Reply:
I just found `check-modified-files-only` overrides `folder-path`, then perhaps we could just check modified files only for release branches as there is no point to update the unmodified files in a release branch if a stale link is detected?

https://github.com/gaurav-nelson/github-action-markdown-link-check/blob/58f84fd654812d0d8da4e4d4a559eda087daf8ce/entrypoint.sh#L155-L202